CrowServe/CrowServe.h

```diff
@@ -65,9 +71,53 @@ class Socket {
         deinit_sockets();
     };
 
-
     bool try_init();
-    void send();
+
+    template <typename Handler>
+    void listen(const Handler& handler) { // todo - include cv for cancel
+        m_thread = std::jthread{[this, handler] {
+            while (true) {
+                Header main_header{};
+                if (receive_into(main_header) < 1) {
+                    // TODO: error handling 0 = conn closed, -1 = check errorno
```

Comment:
Eventually we would return an enum or something, maybe then have a function that converts the enum to const char*, or just return const char*, not sure.

Reply:
oh for the error handling?
 
Since this function runs in a thread I was thinking probably best to handle it here instead of returning anything. That way we don't have to recreate the thread upon reconnect.

If this returns an error we basically need to go into a state where we call `try_init()` repeatedly until we reconnect.

We can do like we did in original crownlink and have a `SocketState` enum which will be `connecting`, `connected`, `failed` etc.

Reply:
Oh sorry, I wasn't specific enough. I was thinking `receive_into` should return the right `enum class` error value or `const char*` string, so we don't have to rely on magic numbers. We would do the error handling here like you are saying :)